wallet-new/test/unit/Test/Spec/Accounts.hs

```diff
@@ -149,16 +166,24 @@ spec = describe "Accounts" $ do
                          -- trying to make one would be overkill.
                          Right _ -> return ()
 
-        prop "Servant handler fails if the parent root doesn't exist" $ withMaxSuccess 50 $ do
+        prop "Servant handler fails if the parent wallet doesn't exist" $ withMaxSuccess 50 $ do
+            monadicIO $ do
+                wId <- pick arbitrary
+                withLayer $ \layer _ -> do
+                    let delete = Handlers.deleteAccount layer wId 100
+                    res <- try . runExceptT . runHandler' $ delete
+                    case res of
+                         Left (_e :: WalletLayer.DeleteAccountError)  -> return ()
+                         Right _ -> fail "Expecting a failure, but the handler succeeded."
+
+        prop "Servant handler fails if the account doesn't exist" $ withMaxSuccess 50 $ do
             monadicIO $ do
                 withFixture $ \_ layer _ Fixture{..} -> do
                     let delete = Handlers.deleteAccount layer (V1.walId fixtureV1Wallet) 100
-                    res <- runExceptT . runHandler' $ delete
+                    res <- try . runExceptT . runHandler' $ delete
                     case res of
-                         Left e  -> fail (show e)
-                         -- There is no Buildable instance for 'NoContent', and
-                         -- trying to make one would be overkill.
-                         Right _ -> return ()
+                         Left (_e :: WalletLayer.DeleteAccountError)  -> return ()
+                         Right _ -> fail "Expecting a failure, but the handler succeeded."
```

Comment:
Let's be more precise here, let's not swallow a potential `Left ServantError` inside this `Right`.